src/node_contextify.cc

```diff
@@ -380,7 +380,16 @@ class ContextifyContext {
     if (ctx->context_.IsEmpty())
       return;
 
-    ctx->sandbox()->Set(property, value);
+    Maybe<bool> is_declared = ctx->global_proxy()->
+        HasRealNamedProperty(ctx->context(), property);
+    bool is_contextual_store = ctx->global_proxy() != args.This();
+
+    bool set_property_will_throw = args.ShouldThrowOnError()
+        && !is_declared.FromJust() && is_contextual_store;
```

Comment:
Minor nit: the readability of this expression would be better if the `FromJust` were to be done above as part of the definition of `is_declared`.